torchinductor/codecache.py

```diff
@@ -68,6 +68,18 @@ def cpp_compiler_search(search):
             continue
     raise exc.InvalidCxxCompiler()
 
+@functools.lru_cache(1)
+def cpp_supports_flag(flag: str) -> bool:
+   rc = subprocess.run([cpp_compiler(), "-E", "-x", "c", flag, "-"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, input="")
+   return rc.returncode == 0
+
+
+
+def cxx_flags() -> str:
+    flags = "-march=native -O3 -ffast-math -fno-finite-math-only"
+    if config.cpp.use_openmp:
+        flags += " -fopenmp"
+    return " ".join([f for f in flags.split() if cpp_supports_flag(f)])
```

Comment:
Do we really want a subprocess run per flag? seems easier to either map it per compiler, or just fail and fallback to running without said flags (as we do today). Thoughts?

Reply:
Hmm, perhaps I don't know where the fallback code is, but `torchdynamo.optimize("inductor", nopython=True)(toy_example)` failed for me with `-march=native` unsupported flag

Also, please note that `cpp_supports_flag` memoizes the answers so it will call subprocess only once.

Reply:
Doesn't matter too much. The fallback would have to be written by you. You can also keep a mapping of compiler->flag, there's not exactly a massive set we support yet. No strong opinion.

Reply:
For `-march=native` I feel we should try to replace that with the correct `-march=...` equivalent for clang.  

On gcc if you remove this flag the generated code won't use AVX vector instructions and be very slow.

Reply:
`march=native` detects the current CPU and uses all instructions available on the detected CPU. If it detects that your CPU supports `avx` it will use them.

Reply:
Well, here is the fun part - clang have `-march=native` on x86, but not on M1 (or probably not on aarch64 linux either)